### PR TITLE
Prevent plugin initialization if Jetpack 8.1 or earlier is installed

### LINKED ISSUE #577: Enforce Jetpack 5.3 minimum
Right now we have text around 5.3 as a minimum version of Jetpack, but we don't enforce it.

I realize we are working to remove the plugin dependency entirely, so let's prioritize this lower for now.

### PR DESCRIPTION
**Pull request by DanReyLop**
Due to changes in the autoloader, if Jetpack 8.1 or earlier is installed alongside WCPay, the whole site would crash. The crash would happen in the `\Jetpack\Config` initialization code.

I added a quick & dirty check to show an admin notice if Jetpack 8.1 or earlier is installed, and prevent the rest of the plugin from initializing. I didn't want to spend a lot of effort in this so it's not as smart as the main dependency checker (which shows a link to update the dependency, tells you which version of the dependency you have, etc). I just wanted something temporary that's better than the site crashing.

Jetpack 8.2 (which is already compatible with WCPay) was released in February 2020. I think we can remove this piece of code in a few months, maybe when that version is 1 year old? Anyway, I left a TODO comment, if someone stumbles upon it in the distant future it can be removed.

To test:
- Install Jetpack 8.1 without using this WCPay branch. Note that the site crashes.
- Install Jetpack 8.2. Everything works fine.
- Install Jetpack 8.1 again, now using this WCPay branch. You'll see a notice telling you to upgrade, and you won't see the `[$] Payments` menu at all.
- Install Jetpack 8.2, still using this WCPay branch. Everything should work fine.
- Disable Jetpack. Everything should still work fine.

Fixes #577.

### CODE REVIEW

File: woocommerce-payments.php [hunk 1]
```diff
@@ -53,3 +56,32 @@ function wcpay_init() {
...
+}
+
+/**
+ * Display an error notice if the installed Jetpack version is too old to even start initializing the plugin.
+ */
+function wcpay_show_old_jetpack_notice() {
+	?>
+	<div class="notice wcpay-notice notice-error">
+		<p><b><?php echo esc_html( __( 'WooCommerce Payments', 'woocommerce-payments' ) ); ?></b></p>
+		<p><?php echo esc_html( __( 'The version of Jetpack installed is too old to be used with WooCommerce Payments. Please deactivate or update Jetpack.', 'woocommerce-payments' ) ); ?></p>
```
Comment (allendav) [inline woocommerce-payments.php:84]: I think it should probably explicitly mention WooCommerce Payments is disabled too.

File: woocommerce-payments.php [hunk 2]
```diff
@@ -53,3 +56,32 @@ function wcpay_init() {
 
 // Make sure this is run *after* WooCommerce has a chance to initialize its packages (wc-admin, etc). That is run with priority 10.
 add_action( 'plugins_loaded', 'wcpay_init', 11 );
```
Comment (allendav) [inline woocommerce-payments.php:58]: Please add a comment saying that if they change the priority here they need to do in wcpay_check_old_jetpack_version too

Comment (DanReyLop) [reply woocommerce-payments.php:59]: Added in https://github.com/Automattic/woocommerce-payments/pull/760/commits/8aa0aa5129af79cc137601d241c5cd28c9d0cdfb

File: woocommerce-payments.php (hunk 1 above)
Comment (DanReyLop) [reply woocommerce-payments.php]: Added in https://github.com/Automattic/woocommerce-payments/pull/760/commits/8aa0aa5129af79cc137601d241c5cd28c9d0cdfb

**Review by dwainm — APPROVED** [review]
🚢

**Comment by dwainm** [comment]
Tested and works as advertised. Also re-ran the build and it passed all non-e2e tests: https://travis-ci.com/github/Automattic/woocommerce-payments/builds/176175825 ( e2e tests were still running when I submitted this comment)

**Comment by dwainm** [comment]
It seems like e2e tests failed but the reasons seem unrelated.

**Pull request merged**